src/main/java/duke/Duke.java

```diff
@@ -134,40 +147,40 @@ public void addTask() throws DukeException {
                 String dateString = dateSplit[0];
                 String timeString = dateSplit[1];
                 LocalDate ld = LocalDate.parse(dateString);
-
                 this.tasks.addTask(taskDesc, convertToTaskType(firstCommand), ld, timeString);
-
-                this.confirmAdditionOfTask();
+                this.writeDataToDuke();
+                return this.confirmAdditionOfTask();
             } else {
                 throw new DukeException("You need to put the date in yyyy-mm-dd hhmm format!");
             }
         } else {
             this.tasks.addTask(taskDesc);
-            this.confirmAdditionOfTask();
+            this.writeDataToDuke();
+            return this.confirmAdditionOfTask();
         }
-        this.writeDataToDuke();
+
     }
 
     /**
      * Confirms the addition of a task.
      */
-    public void confirmAdditionOfTask() {
+    public String confirmAdditionOfTask() {
         int tasksLength = this.tasks.getTasksLength();
```

Comment:
tasksLength may be confusing, might refer to task duration,etc ? Suggest changing variable name to taskListLength